### ISSUE TITLE
Intrinsics tests should use JitLog to check for "no compile" path

### ISSUE THREAD

**Issue by CarolEidt** [dotnet/coreclr#15818]
Many of the SIMD tests (e.g. https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/SIMD/VectorAdd.cs) use the `COMPlus_JitFuncInfoLogFile` capability, exposed by the `JitLog` class in https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/SIMD/VectorUtil.cs, to check that methods that should be recognized by the JIT as intrinsics are *not* compiled.

A similar capability should be used by the HardwareIntrinsics tests.

**Comment by CarolEidt** [comment]
This came up in discussions with @jashook about PR #15784

**Comment by tannergooding** [comment]
FYI. @fiigii, @sdmaclea, @4creators

**Comment by fiigii** [comment]
Sounds good. Can we fold this change into https://github.com/dotnet/coreclr/pull/15771 ?

**Comment by tannergooding** [comment]
> Can we fold this change into #15771 ?
Yes. I'll do that later tonight. I almost have the changes to support templating (using T4 Text Templates) complete as well

**Comment by 4creators** [comment]
It should help a lot

**Comment by fiigii** [comment]
> support templating (using T4 Text Templates) 

Sounds great!

**Comment by 4creators** [comment]
> I almost have the changes to support templating (using T4 Text Templates)

I have done some work on this as well - I have used this to generate calls with constant literals - so most probably we are not overlapping with our changes. For example see:

https://github.com/dotnet/coreclr/pull/15777/files#diff-ef085a1b54ff30e7b69bab877be82801
https://github.com/dotnet/coreclr/pull/15777/files#diff-e5245ce431cffc6dab8a8f2afbbd1234
